docs/source/tutorial_worlds.md

```diff
@@ -13,120 +13,353 @@ section. If you are interested in creating a task, please see
+Some things are automatically shared by our abstractions (like models and
+optimizers), but some users may have to include extra sharing.
+:::
+
+Each clone of a model is relatively cheap to create: the only new memory used
+is the dialogue context. The share object enables us to reuse memory for
+expensive objects, like neural network weights.
+
+At a glance, creating these clones may seem strange. Why do we create all these
+copies? Differentiating between shared and independent state lets us write our
+agents with _independence_ in mind. Agents only need to __focus on a single
+conversation__, maintaining and manipulating the state for just that conversation.
+Every clone will be maintaining _separate conversations_, so when we implement
+a new dataset or model, we only focus on one conversation at a time.
+
+This is what makes up the fundamental backend behind batching or chat services:
+each clone of the agent only has to focus on one conversation at a time, with
+only specific spots for synchronization. In the next section, we'll
+take a look at how this is used in Batching.
+
+## Batching
+
+:::{warning} Already implemented in `TorchAgent`.
+If you're implementing your own custom Neural Network model, you don't need to
+worry about this, except at a high level. This is all handled for you by
+`TorchAgent`.
+:::
+
+Equipped with the ability to create clones, we can use this to implement
+Batching. For a batchsize of 3, we will create 3 clones of the Teacher, and 3
+clones of the Agent. Each of these clones will maintain their own separate,
+independent conversations. Naively, this could be implemented with a simple
+for loop:
+
+```python
+class NaiveBatchWorld(World):
+    def __init__(self, opt, agents):
+        # store the originals
+        self.teacher, self.model = agents
+        # make batchsize copies of all the models
+        self.teacher_copies = []
+        self.model_copies = []
+        self.batchsize = opt['batchsize']
+        for i in range(self.batchsize):
+            self.teacher_copies.append(self.teacher.clone())
+            self.model_copies.append(self.model.clone())
+
+```
+This initialization code is then represented by this graphic:
+
+<center>
+<img src="_static/img/world_share.png" alt="Sharing the teacher and agents" width="50%"/>
+</center>
+
+
+We continue with the implementation of parley:
+
+```python
+    def parley(self):
+        for i in range(self.batchsize):
+            # produce an example dataset item
+            teacher_act = self.teacher_copies[i].act()
+            # perform preprocessing, vectorizing, etc.
+            self.model_copies[i].observe(teacher_act)
+            # produce a model response
+            model_act = self.model_copies[i].act()
+            # compute any metrics of the response
+            self.teacher_copies[i].observe(model_act)   
+```
+
+<center>
+<img src="_static/img/world_batchbasic.png" alt="Independent acts" width="100%"/>
```

Comment:
Could you remove the spell checker line from under the title?

Reply:
Ran a spell check, not what you mean here